Sources/MapboxDirections/Extensions/ForeignMemberContainer.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+import Turf
+
+/**
+ A coding key as an extensible enumeration.
+ */
+struct AnyCodingKey: CodingKey {
+    var stringValue: String
+    var intValue: Int?
+
+    init?(stringValue: String) {
+        self.stringValue = stringValue
+        self.intValue = nil
+    }
+
+    init?(intValue: Int) {
+        self.stringValue = String(intValue)
+        self.intValue = intValue
+    }
+}
+
+extension ForeignMemberContainer {
+    /**
+     Decodes any foreign members using the given decoder.
+     */
+    mutating func decodeForeignMembers<WellKnownCodingKeys>(notKeyedBy _: WellKnownCodingKeys.Type, with decoder: Decoder) throws where WellKnownCodingKeys: CodingKey {
+        let foreignMemberContainer = try decoder.container(keyedBy: AnyCodingKey.self)
+        for key in foreignMemberContainer.allKeys {
+            if WellKnownCodingKeys(stringValue: key.stringValue) == nil {
+                foreignMembers[key.stringValue] = try foreignMemberContainer.decode(JSONValue?.self, forKey: key)
+            }
+        }
+    }
+
+    /**
+     Encodes any foreign members using the given encoder.
+     */
+    func encodeForeignMembers<WellKnownCodingKeys>(notKeyedBy _: WellKnownCodingKeys.Type, to encoder: Encoder) throws where WellKnownCodingKeys: CodingKey {
+        var foreignMemberContainer = encoder.container(keyedBy: AnyCodingKey.self)
+        for (key, value) in foreignMembers {
+            if let key = AnyCodingKey(stringValue: key),
+               WellKnownCodingKeys(stringValue: key.stringValue) == nil {
+                try foreignMemberContainer.encode(value, forKey: key)
+            }
+        }
+    }
+}
+
+/**
+ A GeoJSON *class* that can contain [foreign members](https://datatracker.ietf.org/doc/html/rfc7946#section-6.1) in arbitrary keys.
+ 
+ When subclassing `ForeignMemberClassContainer` type, you should call `decodeForeignMembers(notKeyedBy:with:)` during your `Decodable.init(from:)` initializer if your subclass has added any new properties.
+ */
+public protocol ForeignMemberClassContainer: AnyObject {
```

Comment:
I'm open to naming suggestions here!

Reply:
How about `ForeignMemberContainerClass` or `ForeignMemberContainingClass`?